File: app/src/main/java/mb00/android/codehub/ui/user/inject/UserStarredFragmentModule.java
package mb00.android.codehub.ui.user.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.user.interactor.UserInteractor;
import mb00.android.codehub.ui.user.router.UserRouter;
import mb00.android.codehub.ui.user.viewmodel.UserStarredViewModel;


@Module
public class UserStarredFragmentModule {

    @Provides
    UserStarredViewModel provideUserStarredViewModel(UserInteractor interactor, UserRouter router) {
        return new UserStarredViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/inject/UserStarredFragmentModule.kt
package mb00.android.codehub.ui.user.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.user.interactor.UserInteractor
import mb00.android.codehub.ui.user.router.UserRouter
import mb00.android.codehub.ui.user.viewmodel.UserStarredViewModel


@Module
class UserStarredFragmentModule {

    @Provides
    internal fun provideUserStarredViewModel(interactor: UserInteractor, router: UserRouter): UserStarredViewModel {
        return UserStarredViewModel(interactor, router)
    }

}
